### PR TITLE
Add a styleguide for acceptance tests

### PR DESCRIPTION
**Pull request by tijmenb**
## Context

We have a lot of acceptance tests now, but they're sometimes inconsistent. For example, we have a number of files that contain multiple scenarios ([documented in the tech debt column](https://trello.com/c/13Hf4Ix9/343-multiple-scenarios-in-acceptance-tests-high-urgency)). 

## Changes proposed in this pull request

This adds styleguide with 4 rules and a few examples:

[👉 rendered version](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/testing-styleguide/docs/testing-styleguide.md)

## Guidance to review

Read the rules and see if you agree, understand why they're there, and if they make sense. 

## Link to Trello card

https://trello.com/c/13Hf4Ix9/343-multiple-scenarios-in-acceptance-tests-high-urgency

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: docs/testing-styleguide.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  end
+
+  scenario 'User does a thing' do
+    when_i_press_a_button('Do the thing') # Bad: a parameterised method make the step harder to read
+    then_something_should_happen # Bad: hidden in a module
+  end
+
+  # Bad: multiple scenarios clutter the file and slow down the test suite
+  scenario 'User does a different thing' do
+    when_i_press_a_button('Do the thing') # Bad: a parameterised method make the step harder to read
```
Comment (stevehook) [inline docs/testing-styleguide.md]:
```suggestion
    when_i_press_a_button('Do the thing') # Bad: a parameterised method makes the step harder to read
```

File: docs/testing-styleguide.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  let(:user) { create(:user) } # Bad: a `let` block adds noise to the file and adds indirection
+
+  # Bad: a `before` block adds noise to the file and can make it unclear why something is set up
+  before do
+    sign_in_user(user)
+  end
+
+  scenario 'User does a thing' do
+    when_i_press_a_button('Do the thing') # Bad: a parameterised method make the step harder to read
```
Comment (stevehook) [inline docs/testing-styleguide.md]:
```suggestion
    when_i_press_a_button('Do the thing') # Bad: a parameterised method makes the step harder to read
```

File: docs/testing-styleguide.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
+# Testing styleguide
+
+We use ["Futurelearn style" acceptance tests](https://about.futurelearn.com/blog/how-we-write-readable-feature-tests-with-rspec).
+
+## Rules
```
Comment (chubberlisk) [inline docs/testing-styleguide.md]: Wondering if there should be something about newlines and readability? It's very common that we have multiple given, when, then blocks in the same scenario, if they didn't have a newline between each I would find it hard to read. For example:

```ruby
scenario 'User does a thing' do
  given_i_am_signed_in
  when_i_press_a_button
  then_something_should_happen
  given_i_am_on_this_page
  when_i_do_something
  then_something_else_should_happen
end
```

Vs

```ruby
scenario 'User does a thing' do
  given_i_am_signed_in
  when_i_press_a_button
  then_something_should_happen

  given_i_am_on_this_page
  when_i_do_something
  then_something_else_should_happen
end
```

Comment (tijmenb) [reply docs/testing-styleguide.md]: Added!

**Pull request merged**